cranelift/codegen/src/isa/mod.rs

```diff
@@ -277,6 +277,9 @@ pub trait TargetIsa: fmt::Display + Send + Sync {
     /// will be "labeled" or might have calls between them, typically the number
     /// of defined functions in the object file.
     fn text_section_builder(&self, num_labeled_funcs: u32) -> Box<dyn TextSectionBuilder>;
+
+    /// The function alignment required by this ISA.
+    fn function_alignment(&self) -> u32;
```

Comment:
Can you also use this method in cranelift-object and cranelift-jit?

Reply:
cranelift-object already has support for specifying function alignment as the `function_alignment` field on `ObjectBuilder`, is that sufficient?

cranelift-jit takes an argument that implements `TargetIsa` already, do we need to modify it to ensure that it's emitting aligned function bodies?

Reply:
What do you think about removing the `function_alignment` field from `ObjectBuilder` and calling the `function_alignment()` function in `TargetIsa` instead? Also it looks like we'll need to slightly adapt the `Module` trait to pass through the individual function alignment requirements so that we can pick the largest one when emitting machine code. I've made that change on this branch, but held off on removing `function_alignment` from `ObjectBuilder`.

Looking through cranelift-jit, it seems like it always 16-byte aligns functions, so there might not be any work needed there.